### ISSUE TITLE
Library scanning on docker (on Synology) shows an error message: undefined

### ISSUE THREAD

**Issue by h1p3rcub3** [grimsi/gameyfin#259]
**Describe the bug**
When trying to scan the library (docker on Synology) an "Error while scanning library: undefined" appears. 

**To Reproduce**
1. Install the container vía docker-compose
2. Start the container
3. Access the configuration
4. Press the "Scan library" button

**Expected behavior**
Library scan happens. A toast regarding this event is shown. Games appear on library. No error is shown. 

**Screenshots**
Front toast
![image](https://github.com/grimsi/gameyfin/assets/13078787/a0d9e66e-11c2-4ae7-bed5-47dbee977d9b)

Console output
![image](https://github.com/grimsi/gameyfin/assets/13078787/70384994-d722-4b78-b4f5-0ace1476c4c1)
![image](https://github.com/grimsi/gameyfin/assets/13078787/da1a74fe-13b1-466b-8f42-11de761c6873)

Configuration
![image](https://github.com/grimsi/gameyfin/assets/13078787/070213bc-9b0f-46f6-8ba2-96617ca3d2d7)

**General info (please complete the following information):**
 - Installation method: Docker on Synology DS220+ hardware
 - Browser: Chrome
 - Gameyfin version: 1.4.5

**Additional context**
I tried it on windows directly with the .jar (Bare metal) and it works fine. I was trying to put it in the Synology to have a more consistant experience, as is turned on all day. 

I tried installing with the compose and directly with the registry image, with the same result. 

The "volume1/juegos-lan-party" folder has read/write permissions to "Everyone"

**Comment by grimsi** [comment]
Thanks for your report!
Can you attach the full log as text file? Your screenshots only show the startup of Gameyfin, but not the library scan.

**Comment by h1p3rcub3** [comment]
Thats it. 

The log does not show anything else than "starting scan". Every time I press, a new log line is added with the same content.

![image](https://github.com/grimsi/gameyfin/assets/13078787/15898f2a-3fc6-48aa-9da7-2e8733d2a0d7)

**Comment by h1p3rcub3** [comment]
NAS Hardware: DS220+ (2 Core Celeron J4025, 18Gb of RAM)
OS: Diskstation manager 7.2.1-69057
Docker version: 20.10.23 (API version 1.41)
Java: jdk-21.0.2+13

**Comment by h1p3rcub3** [comment]
I don't know if this is relevant. 

When I compare the Synology container logs with the bare metal in my computer logs I found the following:

In the Synology container, port settings are 8686:8080, but the tomcat uses 8080. 
In the bare metal, when I set a specific port (8686) logs show Tomcart starting on 8686.

I tried adding a "SERVER_PORT" environment variable and the tomcat starts at said server port, but the webpage does not load at all. The container starts, but I'm not able to enter <my-nas-ip>:8686

![image](https://github.com/grimsi/gameyfin/assets/13078787/042f5b85-2114-4a75-83f9-b7afc4a097be)

**Comment by h1p3rcub3** [comment]
Ignore the last comment. I set the container settings to 8686:8686 and SERVER_PORT to 8686. The page now loads but the library scan error message is still there.

**Comment by grimsi** [comment]
To be honest that one is very tricky to debug. I don't have any Synology NAS available for testing and I can't recreate the bug you're describing.
Since Gameyfin is written in Java and you're using Docker to run it there really shouldn't any platform dependent behaviour (apart from file system) affecting Gameyfin.

The logs do not indicate any error (apart from the scan not starting) and the error message in the UI is also very undefined.
The way I see it there is not much more I can do to help you here, I'm sorry.
Your best bet is to find other people with the same bug so we can find the common cause.

Anyway since Gameyfin v2 is in active development (with new backend and new frontend) the bug will most likely be gone once v2 is available. But that will take at least 3 months I would guess.

If you find anything new, just add a comment and I will try to keep an eye on it!

**Comment by h1p3rcub3** [comment]
Hi, 

For anyone with the same issue, I found a solution. Something about the mounting was not working fine. 

What I did is create 3 separate folders, one for the games, one for the cache and another one for the database and mounted them individually.

![image](https://github.com/grimsi/gameyfin/assets/13078787/77f6ce52-790b-45dd-b706-480b28642e3b)

Now I don't get the error. 

Thank you anyway for the help and for sharing with us this cool project.
